frontend/coprs_frontend/coprs/views/apiv3_ns/apiv3_packages.py

```diff
@@ -138,30 +169,57 @@ def get_package_list(ownername, projectname, with_latest_build=False,
     return flask.jsonify(items=items, meta=paginator.meta)
 
 
-@apiv3_ns.route("/package/add/<ownername>/<projectname>/<package_name>/<source_type_text>", methods=POST)
-@api_login_required
-def package_add(ownername, projectname, package_name, source_type_text):
-    copr = get_copr(ownername, projectname)
-    data = rename_fields(get_form_compatible_data(preserve=["python_versions"]))
-    process_package_add_or_edit(copr, source_type_text, data=data)
-    package = PackagesLogic.get(copr.id, package_name).first()
-    return flask.jsonify(to_dict(package))
-
-
-@apiv3_ns.route("/package/edit/<ownername>/<projectname>/<package_name>/<source_type_text>", methods=PUT)
-@api_login_required
-def package_edit(ownername, projectname, package_name, source_type_text=None):
-    copr = get_copr(ownername, projectname)
-    data = rename_fields(get_form_compatible_data(preserve=["python_versions"]))
-    try:
-        package = PackagesLogic.get(copr.id, package_name)[0]
-        source_type_text = source_type_text or package.source_type_text
-    except IndexError:
-        raise ObjectNotFound("Package {name} does not exists in copr {copr}."
-                             .format(name=package_name, copr=copr.full_name))
-
-    process_package_add_or_edit(copr, source_type_text, package=package, data=data)
-    return flask.jsonify(to_dict(package))
+@apiv3_packages_ns.route("/add/<ownername>/<projectname>/<package_name>/<source_type_text>")
+class PackageAdd(Resource):
+    parser = add_package_parser()
+
+    @api_login_required
+    @apiv3_packages_ns.doc(params=add_package_params)
+    @apiv3_packages_ns.expect(parser)
+    @apiv3_packages_ns.marshal_with(package_model)
+    def post(self, ownername, projectname, package_name, source_type_text):
+        """
+        Create a package
+        Create a new package inside a specified Copr project.
+
+        See what fields are required for which source types:
+        https://python-copr.readthedocs.io/en/latest/client_v3/package_source_types.html
+        """
+        copr = get_copr(ownername, projectname)
+        data = rename_fields(get_form_compatible_data(preserve=["python_versions"]))
+        process_package_add_or_edit(copr, source_type_text, data=data)
+        package = PackagesLogic.get(copr.id, package_name).first()
+        return to_dict(package)
+
+
+@apiv3_packages_ns.route("/edit/<ownername>/<projectname>/<package_name>/<source_type_text>")
+class PackageEdit(Resource):
+    parser = edit_package_parser()
+
+    @api_login_required
+    @apiv3_packages_ns.doc(params=edit_package_params)
+    @apiv3_packages_ns.expect(parser)
+    @apiv3_packages_ns.marshal_with(package_model)
+    def post(self, ownername, projectname, package_name, source_type_text=None):
+        """
+        Edit a package
+        Edit an existing package within a Copr project.
```

Comment:
duplicate doc line

Reply:
This was kind of intentional.

![Screenshot_2023-04-27_23-16-48](https://user-images.githubusercontent.com/2151771/234993153-ff07631c-a8b3-484d-ad8f-e2816e6c1559.png)

The first docstring line is used for the "card" header and therefore it is visible even when the route is collapsed. The rest of the docstring is displayed once the route is expanded.

So I basically wanted to write the docstring like

```python
"""
Edit a package
Edit a package within a Copr project.
"""
```

but there is a bug in the docstring parsing - https://github.com/python-restx/flask-restx/pull/536

Anyway, we can write the docstrings in any format you want. My preference would be at least a two-line docstring so that we have at least some text when the route is expanded.